conformance/lang/expressions/is-expr/is_expr.balt

```diff
@@ -0,0 +1,2094 @@
+}
+
+Test-Case: output
+Description: Test string literal as expression in type test expression.
+Labels: boolean, is-expr, string
+
+function init() {
+    boolean b = "A" is string;
+    io:println(b); // @output true
+}
+
+Test-Case: output
+Description: Test byte array literal as expression in type test expression.
+Labels: array-type, boolean, byte, byte-array-literal, is-expr
+
+function init() {
+    boolean b = base64 `aGVsbG8gYmFsbGVyaW5hICEhIQ==` is byte[];
+    io:println(b); // @output true
+}
+
+Test-Case: output
+Description: Test string template expression as expression in type test expression.
+Labels: boolean, is-expr, string, string-template-expr
+
+function init() {
+    boolean b = string `ABC` is string;
+    io:println(b); // @output true
+}
+
+Test-Case: output
+Description: Test xml template expression as expression in type test expression.
+Labels: boolean, is-expr, xml, xml:Element, xml:Text
+
+function init() {
+    boolean b1 = xml `ABC` is xml;
+    io:println(b1); // @output true
+
+    boolean b2 = xml `ABC` is xml:Text;
+    io:println(b2); // @output true
+
+    boolean b3 = xml `<name>Enid Blyton</name><!--Text-->` is xml:Element;
+    io:println(b3); // @output false
+}
+
+Test-Case: output
+Description: Test list constructor expression as expression in type test expression.
+Labels: array-type, boolean, int, is-expr, list-constructor-expr, tuple-type
+
+function init() {
+    boolean b1 = [1, 2, 3] is int[];
+    io:println(b1); // @output true
+
+    boolean b2 = [1, 2, 3] is int[3];
+    io:println(b2); // @output true
+
+    boolean b3 = [1, 2, 3] is [int, int, int];
+    io:println(b3); // @output true
+
+    boolean b4 = [1, 2, 3] is [int...];
+    io:println(b4); // @output true
+
+    boolean b5 = [1, 2, 3] is [int, int...];
+    io:println(b5); // @output true
+
+    boolean b6 = [1, 2, 3] is [int, int, int...];
+    io:println(b6); // @output true
+
+    boolean b7 = [1, 2, 3] is [int, int, int, int...];
+    io:println(b7); // @output true
+}
+
+Test-Case: output
+Description: Test list constructor expression as expression in type test expression.
+Labels: any, array-type, boolean, float, int, is-expr, list-constructor-expr, map-type, record-type, string, tuple-type,
+        union-type
+
+function init() {
+    boolean b1 = [1, "ABC", 10.2, {a: 2}] is any;
+    io:println(b1); // @output true
+
+    boolean b2 = [1, "ABC", 10.2, {a: 2}] is [int, string, float, record {|int a;|}];
+    io:println(b2); // @output true
+
+    boolean b3 = [1, "ABC", 10.2, {a: 2}] is [int, string, float, map<int>];
+    io:println(b3); // @output true
+
+    boolean b4 = [1, "ABC", 10.2, {a: 2}] is [int, string, float|map<int>...];
+    io:println(b4); // @output true
+
+    boolean b5 = [1, "ABC", 10.2, {a: 2}] is (int|string|float|record {|int a;|})[];
+    io:println(b5); // @output true
+
+    boolean b6 = [1, "ABC", 10.2, {a: 2}] is (int|string|float|map<int>)[4];
+    io:println(b6); // @output true
+}
+
```

Comment:
Also, a separate test for semantic subtyping when the list constructor type having never rest descriptor.
ballerina-platform/ballerina-lang#35612

 ```bal 
 [int, string, never...] t1 = [1, "s"];
 io:println(t1 is [int, string]); // @output true
```